includes/class-amp-http.php

```diff
@@ -235,9 +235,15 @@ public static function get_amp_cache_hosts() {
 		 */
 		foreach ( $domains as $domain ) {
 			if ( function_exists( 'idn_to_utf8' ) ) {
-				// The third parameter is set explicitly to prevent issues with newer PHP versions compiled with an old ICU version.
+				$intl_idna_variant = defined( 'INTL_IDNA_VARIANT_UTS46' ) ? INTL_IDNA_VARIANT_UTS46 : false;
```

Comment:
I've cherry-picked this change into https://github.com/ampproject/amp-wp/pull/6600.